doc/howto/setting-up-logging.rst

```diff
@@ -0,0 +1,123 @@
+===============================
+Controlling log output from NAV
+===============================
+
+Overview
+========
+
+All NAV subsystems will produce logs about what's going on at any given time.
+By default, all NAV daemons will send all log records to the file system.  Log
+files from individual subsystems are created in the directory configured by the
+``LOG_DIR`` configuration option in :file:`nav.conf`.
+
+The notable exception to this rule is NAV's web subsystem, which is defined as
+a WSGI application.  The WSGI application usually runs inside a web server or a
+separate WSGI application container, and will by default print all log lines to
+its standard error file descriptor.  It is entirely up to the web/application
+server to decide where this output is directed (in a typical Apache setup with
+``mod_wsgi``, the logs are sent to Apache's defined error log file for the
+specific ``VirtualHost``).
+
+Controlling log levels
+======================
+
+NAV employs `Python's standard logging facility
+<https://docs.python.org/3/library/logging.html>`_, and utilizes a hierarchy of
+log handlers for different parts of the codebase.  Most NAV log handlers will
+be named after the Python module that uses it, meaning that the log handler
+hierarchy will usually correspond to the Python module hierarchy rooted at
+``nav``.
+
+All emitted log records will have one of 5 different log levels: **DEBUG**,
+**INFO**, **WARNING**, **ERROR** or **CRITICAL**.  NAV's default logging configuration
+will only emit log records of level **INFO** or higher.
+
+For most NAV usage, leaving the level at **INFO** is fine.  If, however, you need
+to debug a problem or at least get more details out of a specific NAV
+subsystem, the log levels of the log handler hierarchy can be controlled
+through the configuration file :file:`logging.conf`, whose defaults look like this:
+
+   .. literalinclude:: ../../python/nav/etc/logging.conf
+
+As you can see, this sets the log level of NAV's top-level logger ``nav`` to
+*INFO*.  Since the loggers operate in a hierarchy, this level now also applies
+to loggers like ``nav.ipdevpoll`` or ``nav.eventengine.queue`` (unless they
+have been configured with their own explicit log levels).
+
+If you wish to enable **DEBUG** level logging, it's usually *not recommended*
+to set ``nav = DEBUG`` in the ``[levels]`` section: This will cause all parts
+of NAV to log in extreme detail, in all log files.  Rather, if you wish to
+specifically have debug logging from :program:`ipdevpoll`, you can set
+``ipdevpoll = DEBUG``.  This maybe OK if you don't know which part of
+:program:`ipdevpoll` you are interested in debugging, but it is still going to
+be extremely verbose.  If you only wish to get debug details on which plugins
+are selected and run for single jobs, you might want to only set
+``nav.ipdevpoll.jobs.jobhandler = DEBUG``.
+
```

Comment:
``ipdevpoll = DEBUG``, but ``nav.ipdevpoll.jobs.jobhandler = DEBUG``, isn't that inconsistent? Should the first be ``nav.ipdevpoll = DEBUG``?

Reply:
Indeed, `ipdevpoll` won't do it, it should `nav.ipdevpoll` :)